pcp/Platform.c

```diff
@@ -248,6 +248,25 @@ static const char* Platform_metricNames[] = {
    [PCP_METRIC_COUNT] = NULL
 };
 
+#ifndef HAVE_PMLOOKUPDESCS
+int pmLookupDescs(int numpmid, pmID *pmids, pmDesc *descs) {
+   for (int i = 0; i < numpmid; i++) {
+      /* expect some metrics to be missing - e.g. PMDA not available */
+      if (pmids[i] == PM_ID_NULL)
+         continue;
+
+      int sts = pmLookupDesc(pmids[i], &descs[i]);
+      if (sts < 0) {
+         if (pmDebugOptions.appl0)
+            fprintf(stderr, "Error: cannot lookup metric %s(%s): %s\n",
+                    pcp->names[i], pmIDStr(pcp->pmids[i]), pmErrStr(sts));
+         pmids[i] = PM_ID_NULL;
+      }
+   }
+   return 0;
```

Comment:
This completely hides when all metrics failed to resolve. Shouldn't we at least check that at least one metric could be resolved in order to return "success"? What's the PCP API behaviour for partial success?

Reply:
Depends on the API, but in this case using metric identifier PM_ID_NULL indicates failure at some point of the lookup path (metric name lookup has happened just prior to this descriptor lookup).  I'll add some logic to handle the case where we end up with no metrics available across all of htop - as a final sanity check during startup.